presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -562,6 +583,26 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
         return client;
     }
 
+    private EncryptionMaterialsProvider createEncryptionMaterialsProvider(Configuration hadoopConfig)
+    {
+        EncryptionMaterialsProvider emp = null;
+        String empClassName = hadoopConfig.get(S3_ENCRYPTION_MATERIALS_PROVIDER);
```

Comment:
Configuring this via the Hadoop config system doesn't fit with the way Presto and the Hive connector do configuration. We have first class config support in `HiveClientConfig` for most everything that a user would want to configure. All of these `S3_` Hadoop config variables are for passing the Hive connector's configuration into `PrestoS3FileSystem` (because it is created by Hadoop and thus we don't have direct access to configure it). See `HdfsConfigurationUpdater` for where these are propagated from `HiveClientConfig` into the Hadoop `Configuration` object.